specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json

```diff
@@ -0,0 +1,9330 @@
+        "operationId": "Activate_Promotion",
+        "x-ms-examples": {
+          "PromotionActivation": {
+            "$ref": "./examples/PromotionActivation.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/promotionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "tags": [
+          "Promotion, Activate"
+        ],
+        "responses": {
+          "200": {
+            "description": "The resource has been created",
+            "schema": {
+              "$ref": "#/definitions/PromotionResponse"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "get": {
+        "summary": "Get a specific Promotion.",
+        "description": "Get the details of the `Promotion`.",
+        "operationId": "Promotion_Get",
+        "x-ms-examples": {
+          "GetPromotion": {
+            "$ref": "./examples/PromotionDetails.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/promotionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "tags": [
+          "Promotion"
+        ],
+        "responses": {
+          "200": {
+            "description": "Get the details of the `Promotion`.",
+            "schema": {
+              "$ref": "#/definitions/PromotionResponse"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/promotions": {
+      "get": {
+        "summary": "Get all `Promotion`s.",
+        "description": "List of all the `Promotion`s that the user has access.",
+        "operationId": "Promotions_List",
+        "x-ms-examples": {
+          "PromotionList": {
+            "$ref": "./examples/GetPromotions.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "tags": [
+          "Promotion"
+        ],
+        "responses": {
+          "200": {
+            "description": "List of `Promotion`s",
+            "schema": {
+              "$ref": "#/definitions/PromotionList"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Billing/promotions/{promotionSkuId}": {
```

Comment:
this way to model eligibility of a subscription is weird and can be confusing.  It looks like a GET API but if the subscription segment is added to the path it transitions to a check eligibility API.
Better way to model will be to take subscription Id in the body. And may be make it a POST /providers/Microsoft.Billing/promotions/{id}/checkEligibility

Reply:
The reason is we want to check the eligibility on 1000s of such subscriptions that a user may have and leverage the batch semantics that are available only on get.

Reply:
Discussed offline added the /checkEligibility